File: 백준/Silver/14916. 거스름돈/거스름돈.py
import sys

change = [-1 for i in range(100001)]
change[0] = 0

money = [2, 5]
for i in range(2, 100001):
    tmp = []
    for j in range(2):
        if i-money[j] < 0:
            continue
        if change[i-money[j]] == -1:
            continue
        tmp.append(change[i-money[j]]+1)
    if len(tmp) == 0:
        change[i] = -1
    else:
        change[i] = min(tmp)


n = int(sys.stdin.readline())

print(change[n])

File: 프로그래머스/3/12907. 거스름돈/거스름돈.java
class Solution {
    public int solution(int n, int[] money) {
        int[] dp = new int[n+1];
        for(int m: money){
            dp[m] += 1;
            for(int i = 0; i<=n; i++){
                if(i-m>=0){
                    dp[i] += dp[i-m];
                    dp[i] %= 1_000_000_007;
                }
            }
        }
        return dp[n];
    }
}